### PR TITLE
adding default font to website#1657

### LINKED ISSUE #1657: Add a default font for the platform
**Describe the bug**
Currently we are not defining a font to be used on the platform so the browser resorts to using the browser-default font. This creates inconsistency when using the platform on different browsers.

**Expected behavior**
Add a font to the platform (personally prefer "Inter", finalise with @nitin10s) using a OTF/TTF file via the Asset Pipeline

**Are you working on this issue? (Yes/No)**
No

### PR DESCRIPTION
**Pull request by pavanjoshi914**
Fixes #
closes #1657 
#### Describe the changes you have made in this PR -
    default font -"inter" is been implement via asset pipeline 
### Screenshots of the changes (If any) -
current font which can vary with browsers
before
default font Inter after implementation of code
after

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: app/assets/stylesheets/layout.scss [hunk 1]
```diff
@@ -1,3 +1,8 @@
+@font-face {
+  font-family: 'Inter';
+  src: url(asset-path("Inter.ttf")) format("truetype");
```
[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply app/assets/stylesheets/layout.scss]: resolved

[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply app/assets/stylesheets/layout.scss]: resolved

[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply app/assets/stylesheets/layout.scss]: resolved

File: app/assets/stylesheets/layout.scss [hunk 2]
```diff
@@ -1,3 +1,10 @@
+@font-face {
+  font-family: 'Inter';
+  src: url(asset-path('Inter.ttf')) format('truetype');
```
Comment (aayush-05) [inline app/assets/stylesheets/layout.scss:3]:
```suggestion
  src: font-url('Inter.ttf') format('truetype');
```

File: config/application.rb [hunk 3]
```diff
@@ -26,5 +26,6 @@ class Application < Rails::Application
     config.site_image = "https://circuitverse.org/img/circuitverse2.svg"
     config.site_description = "Explore Digital circuits online with CircuitVerse. With our easy to use simulator interface, you will be building circuits in no time."
     config.slack_url = "https://join.slack.com/t/circuitverse-team/shared_invite/enQtNjc4MzcyNDE5OTA3LTdjYTM5NjFiZWZlZGI2MmU1MmYzYzczNmZlZDg5MjYxYmQ4ODRjMjQxM2UyMWI5ODUzODQzMDU2ZDEzNjI4NmE"
+    config.assets.paths << Rails.root.join("app", "assets", "fonts")
```
Comment (satu0king) [inline config/application.rb:29]: Is this needed?

File: app/assets/stylesheets/layout.scss (hunk 2 above)
Comment (satu0king) [reply app/assets/stylesheets/layout.scss:3]: Shouldn't it be `fonts/Inter.tff` instead of just `Inter.tff` ?

**Review by satu0king — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/layout.scss (hunk 2 above)
Comment (pavanjoshi914) [reply app/assets/stylesheets/layout.scss:3]: using asset-path() inside url() or directly using font-url() both works fine 
we have to keep path only as 'Inter.ttf'... if we  use fonts/Inter.ttf we get path such as ' localhost:3000/fonts/fonts/Inter.ttf'
and we end with 404 error as font-url() will take care of fonts folder we just need 'Inter.ttf' as path...

File: config/application.rb (hunk 3 above)
Comment (pavanjoshi914) [reply config/application.rb:29]: this line add's the new created folder 'fonts' to the asset path in config/application.rb (it is always better to do that)
but things work correctly without it also ,If you wish I will remove it

**Review by aayush-05 — APPROVED** [review]
LGTM

File: config/application.rb (hunk 3 above)
Comment (satu0king) [inline config/application.rb:29]:
```suggestion
```

Comment (pavanjoshi914) [reply config/application.rb]: Yes, I appreciate that,  if we implement fonts  through vendor folder then we might need extra such line
```
app.config.assets.paths << Rails.root.join('vendor', 'assets', 'fonts')  
```
hence removed it ...
also  possible config I think we can add in assets.rb is just
```
app.config.assets.precompile << /\.(?:svg|eot|woff|ttf)$/
```
which is already present I, if any additional configs required then let me know

**Pull request merged**